packages/sdk/src/types.ts

```diff
@@ -60,41 +91,59 @@ export type MakeBuyOrderParams = MakeSellOrderParams;
 export enum MarketplaceName {
   Opensea = "opensea",
   Trader = "trader",
-  Looksrare = "looksrare",
+  LooksRare = "looksrare",
 }
 
-interface Opensea {
-  marketplaceName: MarketplaceName.Opensea;
+export interface NormalizedOrder<OriginalOrder> {
+  id: string;
+  makerAssets: Asset[];
+  takerAssets: Asset[];
+  maker: string;
+  originalOrder: OriginalOrder;
 }
 
-interface Trader {
-  marketplaceName: MarketplaceName.Trader;
-}
+export type OpenseaOrder = NormalizedOrder<OpenseaOriginalOrder>;
 
-interface Looksrare {
-  marketplaceName: MarketplaceName.Looksrare;
-}
+export type TraderOrder = NormalizedOrder<TraderOriginalOrder>;
+
+export type LooksRareOrder = NormalizedOrder<LooksRareOriginalOrder>;
 
-export interface OpenseaOrder extends Opensea {
-  marketplaceOrder: _OpenseaOrder;
+interface ResponseBase<N extends MarketplaceName> {
+  marketplaceName: N;
 }
 
-export interface TraderOrder extends Trader {
-  marketplaceOrder: PostOrderResponsePayload;
+interface ResponseData<N extends MarketplaceName, D> extends ResponseBase<N> {
+  data: D;
 }
 
-export interface LooksRareOrder extends Looksrare {
-  marketplaceOrder: _LooksRareOrder;
+interface ResponseError<N extends MarketplaceName> extends ResponseBase<N> {
+  error: {
+    message: string;
+  };
 }
 
-export type Order = OpenseaOrder | TraderOrder | LooksRareOrder;
+type Response<N extends MarketplaceName, D> = Merge<
+  ResponseData<N, D>,
+  ResponseError<N>
+> &
+  ResponseBase<N>;
```

Comment:
Note: the `&  ResponseBase<N>` is required to make `markplaceName` field a guaranteed one 😞